### PR TITLE
[Feature] Add learning rate scheduler in FS

### PR DESCRIPTION
**Pull request by DavdGao**
As the title says, add the support for learning rate scheduler

### CODE REVIEW

File: federatedscope/core/configs/cfg_training.py [hunk 1]
```diff
@@ -22,6 +22,10 @@ def extend_training_cfg(cfg):
     cfg.train.optimizer.type = 'SGD'
     cfg.train.optimizer.lr = 0.1
 
+    # you can add new arguments 'aa' by `cfg.train.scheduler.aa = 'bb'`
+    cfg.train.scheduler = CN(new_allowed=True)
```
Comment (xieyxclack) [inline federatedscope/core/configs/cfg_training.py:26]: Why we need to set `new_allowed=True` here?

**Review by xieyxclack — COMMENTED** [review]
LGTM

File: federatedscope/core/configs/cfg_training.py (hunk 1 above)
Comment (DavdGao) [reply federatedscope/core/configs/cfg_training.py:26]: > Why we need to set `new_allowed=True` here?

It allow the users set different lr schedulers, and we don't need to list all the potential parameters. For example, if the type of lr scheduler is `StepLR`(torch.optim.lr_scheduler.StepLR), users can add their parameters as follows 
```
train:
    scheduler:
        type: StepLR
        step_size: 5
        gamma: 0.1
```

**Review by xieyxclack — APPROVED** [review]
